flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/aggregate/asyncprocessing/AsyncStateGroupAggFunction.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.flink.table.runtime.operators.aggregate.utils.GroupAggHelper;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.util.Collector;
+
+/** Aggregate Function used for the groupby (without window) aggregate with async state api. */
+public class AsyncStateGroupAggFunction extends GroupAggFunctionBase {
+
+    private static final long serialVersionUID = 1L;
+
+    // stores the accumulators
+    private transient ValueState<RowData> accState = null;
+
+    private transient AsyncStateGroupAggHelper aggHelper = null;
+
+    /**
+     * Creates a {@link GroupAggFunction}.
+     *
+     * @param genAggsHandler The code generated function used to handle aggregates.
+     * @param genRecordEqualiser The code generated equaliser used to equal RowData.
+     * @param accTypes The accumulator types.
+     * @param indexOfCountStar The index of COUNT(*) in the aggregates. -1 when the input doesn't
+     *     contain COUNT(*), i.e. doesn't contain retraction messages. We make sure there is a
+     *     COUNT(*) if input stream contains retraction.
+     * @param generateUpdateBefore Whether this operator will generate UPDATE_BEFORE messages.
+     * @param stateRetentionTime state idle retention time which unit is MILLISECONDS.
+     */
+    public AsyncStateGroupAggFunction(
+            GeneratedAggsHandleFunction genAggsHandler,
+            GeneratedRecordEqualiser genRecordEqualiser,
+            LogicalType[] accTypes,
+            int indexOfCountStar,
+            boolean generateUpdateBefore,
+            long stateRetentionTime) {
+        super(
+                genAggsHandler,
+                genRecordEqualiser,
+                accTypes,
+                indexOfCountStar,
+                generateUpdateBefore,
+                stateRetentionTime);
+    }
+
+    @Override
+    public void open(OpenContext openContext) throws Exception {
+        super.open(openContext);
+
+        final StreamingRuntimeContext runtimeContext =
+                (StreamingRuntimeContext) getRuntimeContext();
+
+        InternalTypeInfo<RowData> accTypeInfo = InternalTypeInfo.ofFields(accTypes);
+        ValueStateDescriptor<RowData> accDesc = new ValueStateDescriptor<>("accState", accTypeInfo);
+        if (ttlConfig.isEnabled()) {
+            accDesc.enableTimeToLive(ttlConfig);
+        }
+
+        accState = runtimeContext.getValueState(accDesc);
+        aggHelper = new AsyncStateGroupAggHelper();
+    }
+
+    @Override
+    public void processElement(RowData input, Context ctx, Collector<RowData> out)
+            throws Exception {
+        RowData currentKey = ctx.getCurrentKey();
+        accState.asyncValue()
+                .thenAccept(acc -> aggHelper.processElement(input, currentKey, acc, out));
+    }
+
+    private class AsyncStateGroupAggHelper extends GroupAggHelper {
```

Comment:
Actually we can reuse one unified GroupAggHelper to replace the current `AsyncStateGroupAggHelper` and  `SyncStateGroupAggHelper` (since they two have no different logic)？

Reply:
There are mainly two differences:
1. the ValueState in `SyncStateGroupAggHelper` is `org.apache.flink.api.common.state.ValueState` and that in `AsyncStateGroupAggHelper ` is `org.apache.flink.api.common.state.v2.ValueState`
2. I'll update the logic in `AsyncStateGroupAggHelper` to `asyncUpdate` and `asyncClear`